app/lib/linters/crunch_api.rb

```diff
@@ -10,77 +11,56 @@ def initialize(file:, base_url: ENV['CRUNCH_BASE_URL'])
       @base_url = base_url
     end
 
+    # Post the file to 42Crunch according to their API v1 "Create an API (from file)" spec
+    # https://www.postman.com/get-42crunch/workspace/42crunch-api/request/13761657-335bcb36-c68f-43a1-823d-741b62b55bc6
+    def create_api
+
+      RestClient.post(URI.join(@base_url, "/api/v1/apis").to_s,
+      { # Body of the request
+        cid: ENV['COLLECTION_ID'], # Collection id, returned by "Create a collection"
+        name: File.basename(file.path, ".*"), # API Display Name
+        yaml: false, # Set to true if the specification file was converted to JSON from YAML
+        specfile: file # Raw OAS file in JSON format - YAML is not supported
+      },
+      { # Request headers
+      'X-API-KEY': ENV['CRUNCH_API_KEY']
+      })
+
+    end
+
+    # Use the created API ID to get the linting results from 42Crunch using their
+    # API V1 "Retrieve and visualize a security audit report" spec
+    # https://www.postman.com/get-42crunch/workspace/42crunch-api/request/13761657-85a4551d-8350-4744-ba62-e4289103ec81
+    # @param id [String] the ID of the report to retrieve
+    # @return [String] the raw JSON response
+    def retrieve_report(id)
+      url = URI.join(@base_url, "/api/v1/apis/", "#{id}/", "assessmentreport").to_s
+      RestClient.get(url,
+      { # Request headers
+      'X-API-KEY': ENV['CRUNCH_API_KEY']
+      })
+    end
+
     # Lints an unploaded file with 42crunch.
     #
     # @param file [File] the file to be linted
     # @return [String] the raw JSON response
     def lint_to_json
 
-      # Post the file to 42Crunch according to their API v1 "Create an API (from file)" spec
-      # https://www.postman.com/get-42crunch/workspace/42crunch-api/request/13761657-335bcb36-c68f-43a1-823d-741b62b55bc6
-      begin
-        create_response = RestClient.post(URI.join(@base_url, "/api/v1/apis").to_s,
-        { # Body of the request
-          cid: ENV['COLLECTION_ID'], # Collection id, returned by "Create a collection"
-          name: File.basename(file.path, ".*"), # API Display Name
-          yaml: false, # Set to true if the specification file was converted to JSON from YAML
-          specfile: file # Raw OAS file in JSON format - YAML is not supported
-        },
-        { # Request headers
-        'X-API-KEY': ENV['CRUNCH_API_KEY']
-        })
-      rescue RestClient::MovedPermanently,
-             RestClient::Found,
-             RestClient::TemporaryRedirect => err
-        err.response.follow_redirection
-      rescue RestClient::Unauthorized, RestClient::Forbidden => err
-        raise 'Access denied by 42Crunch API'
-      rescue RestClient::RequestFailed => err
-        raise '42Crunch Create Request Failed'
-      end
+      # Create an API for the file we initialized this class with then
+      # parse the JSON response to get the ID needed to get the linting results
+      created_api_id = JSON.parse(create_api())["desc"]["id"]
```

Comment:
I'd suggest [using `dig`](https://apidock.com/ruby/Hash/dig) as it fails a little more consistently that chaining a series of `[]` calls. So:

```ruby
created_api_id = JSON.parse(create_api()).dig("desc","id")
```

Reply:
Good shout. Changed to use dig